vxquery-core/src/main/java/org/apache/vxquery/runtime/functions/index/VXQueryIndexReader.java

```diff
@@ -208,12 +205,12 @@ private int buildElement(ArrayBackedValueStorage abvsFileNode, int fieldNum) thr
             this.firstElement = false;
             firstFinish = whereIFinish - this.childSequenceTypes.size() + 1;
             String firstBit = contents.substring(1, firstColon);
-            List<String> names = new ArrayList<String>();
-            List<String> values = new ArrayList<String>();
-            List<String> uris = new ArrayList<String>();
-            List<String> localNames = new ArrayList<String>();
-            List<String> types = new ArrayList<String>();
-            List<String> qNames = new ArrayList<String>();
+            List<String> names = new ArrayList<>();
```

Comment:
Can these ArrayLists be created once and reused on each call? Basically clear or reset the array each time?